src/main/java/ognl/DefaultClassResolver.java

```diff
@@ -41,28 +42,30 @@
  */
 public class DefaultClassResolver extends Object implements ClassResolver
 {
-    private Map     classes = new HashMap(101);
+    private Map     classes = new ConcurrentHashMap(101);
 
     public DefaultClassResolver()
     {
         super();
     }
 
-	public Class classForName(String className, Map context) throws ClassNotFoundException
-	{
-	    Class       result = null;
+    public Class classForName(String className, Map context) throws ClassNotFoundException
+    {
+        Class       result = null;
 
         if ((result = (Class)classes.get(className)) == null) {
-    		try {
-    		    result = Class.forName(className);
-    		} catch (ClassNotFoundException ex) {
-    			if (className.indexOf('.') == -1) {
-    			    result = Class.forName("java.lang." + className);
-        			classes.put("java.lang." + className, result);
-        		}
-    		}
-			classes.put(className, result);
-    	}
-	    return result;
-	}
+            try {
+                result = Class.forName(className);
+            } catch (ClassNotFoundException ex) {
+                if (className.indexOf('.') == -1) {
```

Comment:
I think when we are here, this `if` clause can be moved out of the catch and implemented as follow (`classForName` already throws `ClassNotFoundException`):

```java
if (className.indexOf('.') == -1) {
    result = Class.forName("java.lang." + className);
    classes.put("java.lang." + className, result);
} else {
    classes.put(className, Class.forName(className));
}
```

then `result` and additional check for null is not needed.

Reply:
I refactored this code with your suggestion as reference.